linchpin/__init__.py

```diff
@@ -1,386 +1,497 @@
-                records for any/all targets in the RunDB will be displayed.
+            rundb.update_record(target,
+                                rundb_id,
+                                'inputs',
+                                [
+                                    {'topology_data':
+                                     provision_data[target]['topology']}
+                                ])
 
-    fields:     Comma separated list of fields to show in the display.
-    (Default: action, uhash, rc)
 
-    (available fields are: uhash, rc, start, end, action)
 
-    """
+            if provision_data[target].get('layout', None):
+                self.set_evar('layout_data', provision_data[target]['layout'])
 
-    all_fields = json.loads(lpcli.get_cfg('lp', 'rundb_schema')).keys()
+                rundb.update_record(target,
+                                    rundb_id,
+                                    'inputs',
+                                    [
+                                        {'layout_data':
+                                         provision_data[target]['layout']}
+                                    ])
 
-    if not fields:
-        fields = ['action', 'uhash', 'rc']
-    else:
-        fields = fields.split(',')
+        # set inventory_file
 
-    invalid_fields = [field for field in fields if field not in all_fields]
+        # set the current target data
+        # this will differ with dynamic inputs
+#        self.target_data = pf[target]
+#        self.target_data["extra_vars"] = self.get_evar()
 
-    if invalid_fields:
-        ctx.log_state('The following fields passed in are not valid: {0}'
-                      ' \nValid fields are {1}'.format(fields, all_fields))
-        sys.exit(89)
+        # note : changing the state triggers the hooks
+#        self.hooks.rundb = (rundb, rundb_id)
+#        self.pb_hooks = self.get_cfg('hookstates', action)
+            self.ctx.log_debug('calling: {0}{1}'.format('pre', action))
 
-    no_records = []
-    output = 'run_id\t'
+#        if 'pre' in self.pb_hooks:
+#            self.hook_state = '{0}{1}'.format('pre', action)
 
-    for f in fields:
-        output += '{0:>10}\t'.format(f)
+        # FIXME need to add rundb data for hooks results
 
-    output += '\n'
-    output += '--------------------------------------------------'
+        # invoke the appropriate action
+            return_code, results[target]['task_results'] = (
+                self._invoke_playbooks(resources, action=action,
+                                       console=ansible_console)
+            )
 
+            if not return_code:
+                self.ctx.log_state("Action '{0}' on Target '{1}' is "
+                                   "complete".format(action, target))
 
-    try:
-        journal = lpcli.lp_journal(targets=targets, fields=fields, count=count)
+        # FIXME Check the result[target] value here, and fail if applicable.
```

Comment:
I think this is related to RFE #339

Reply:
This will get addressed in a later release.